src/Nix/Builtins.hs

```diff
@@ -750,7 +751,8 @@ replaceStrings tfrom tto ts =
                             , Builder.singleton h
                             ]
                     _ -> go rest $ result <> Builder.fromText replacement
-        toNix $ principledModifyNixContents (\x -> go x mempty) ns
+        let ctx = HS.unions $ principledGetContext ns : map principledGetContext nsTo
```

Comment:
Not quite right... We only want to add in the context if the replacement actually happens.

Reply:
Fixed.